Jenkins.io - add image guidelines to contributing.adoc (#6239)
jenkins-infra/jenkins.io: CONTRIBUTING.adoc

. *Extend Jenkins Documentation* - this documentation is for people who want to _extend_ the functionality of Jenkins by developing their own Jenkins plugins.
Like the Jenkins User Documentation (above), the content is based on the same link:https://www.divio.com/blog/beginners_guide_to_documentation/[documentation model].
The content for this set of documentation is written up as a combination of `.haml` and `.adoc` files located in the link:content/doc/developer[`content/doc/developer/`] directory.
Read more about adding pages to this documentation in <<adding-a-stand-alone-page,Adding a stand-alone-page>>.
. *Solution pages* - topic-specific destination pages providing a high-level overview of a topic with links into getting started guides, handbook chapters, relevant plugins and multimedia related to the topic.
Be aware that some of this content might already be present in the Jenkins User / Extend Jenkins Documentation.

The documentation pages can use the same metadata (`title`, `description`, `opengraph:image`) as blog posts.

==== Adding images

> When adding screenshots or images to documentation, there are methods to ensure that the images are focused, clear, and useful to the reader:

Reviewer: ```suggestion
When you add screenshots or images to documentation, there are methods to ensure that the images are focused, clear, and useful to the reader:
```

* *User Handbook* - rich and in-depth documentation, separated into chapters, each of which covers a given topic/feature of Jenkins.
This is conceptually and structurally similar to the link:https://www.freebsd.org/doc/en_US.ISO8859-1/books/handbook/[FreeBSD Handbook].
The User Handbook covers the fundamentals on how to use Jenkins, as well as content which is not explained in the *Tutorials* or *How-to Guides*.
This content is based predominantly on the "technical reference" description in Michael Nicholson's blog post "link:https://www.divio.com/blog/beginners_guide_to_documentation/[Beginner's Guide to Documentation: Here's What You Need to Know]", with appropriate "discussion" (background/overview material) and general "how-to guide" (specific to the chapter/topic in question) material.
Read more about <<adding-user-handbook-content,Adding User Handbook content>>.
* *Resources*:
** The Pipeline Syntax Reference is a link to the published link:content/doc/book/pipeline/syntax.adoc[syntax.adoc] reference page in the *User Handbook*.
** The Pipeline Steps Reference consists of Asciidoc files which are auto-generated from content within the relevant Pipeline plugin source code.

> Therefore, to contribute to this content, you need to edit the relevant plugin's source code.

Reviewer: ```suggestion
Therefore, to contribute to this content, you must edit the relevant plugin's source code.
```

* *Recent Tutorial Blog Posts* - these are a list of the most recently published blog posts presented as tutorials (and tagged with the *tutorial* tag).

> * *Guided Tour* (Deprecated) - This part of the documentation is being decommissioned in favor of the *Tutorials* and *How-to guides* parts, both of which focus more on teaching people how to use Jenkins or helping people with specific use-cases.

Reviewer: ```suggestion
* *Guided Tour* (Deprecated) - This part of the documentation is being decommissioned in favor of the *Tutorials* and *How-to guides* parts, both of which focus more on teaching how to use Jenkins or helping with specific use-cases.
```

If an `.adoc` file name begins with a underscore (e.g. link:content/doc/tutorials/_prerequisites.adoc[`content/doc/tutorials/_prerequisites.adoc`]), this means that the content is used as an link:https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#include-files[Asciidoc inclusion] on another page.

==== Adding a How-to guide page

This section will be completed when the first (or first set of) "how-to guides" are written.

==== Adding User Handbook content

The different chapters for the Handbook are located in the link:content/doc/book[`content/doc/book/`] directory.

To add a chapter:

. Add a new subdirectory (within this directory) whose name reflects your chapter title.
. Specify this subdirectory's name as a new entry in the link:content/doc/book/_book.yml[`content/doc/book/_book.yml`] file.
The position of the entry in this file determines the order in which the chapter appears in the User Handbook.
. Create an `index.adoc` file within the subdirectory you created above.
Feel free to copy another chapter's `index.adoc` content as a template/starting point.

> The content on this page should be an overview ("discussion" material) about the subject of this chapter, such as some big new Jenkins feature. +

Reviewer: + may need to be on a new line?

> Once you do this, the chapters will automatically surface on the User Handbook home page (provided by link:content/doc/book/index.html.haml[`content/doc/book/index.html.haml`]), which will automatically appear https://jenkins.io/doc/book/[here] and in the TOC on the left of this page when accepted. +

Reviewer: Same here possibly?

> Once you do add some topics to this chapter page, as well as additional pages of topics within a chapter as described below, it's recommended that you link to these topics from within the overview to help readers find this information.

Reviewer: ```suggestion
Once you do add some topics to this chapter page, as well as additional pages of topics within a chapter as described below, we recommend that you link to these topics from within the overview to help readers find this information.
```

To add a page (i.e. "section") within a chapter:

> . Within the relevant chapter subdirectory, create a new `.adoc` file whose name reflects your page title. Feel free to copy another section's `.adoc` content as a template/starting point.

Reviewer: ```suggestion
. Within the relevant chapter subdirectory, create a new `.adoc` file whose name reflects your page title. You can copy another section's `.adoc` content as a template/starting point.
```

> . Specify this `.adoc` file's name as a new entry in a `_chapter.yml` file within this directory. Feel free to copy an empty `_chapter.yml` file from another subdirectory/chapter (e.g. from the `glossary` directory).

Reviewer: ```suggestion
. Specify this `.adoc` file's name as a new entry in a `_chapter.yml` file within this directory. You can copy an empty `_chapter.yml` file from another subdirectory/chapter (e.g. from the `glossary` directory).
```

> The position of the entry in this file determines the order in which the page appears within the chapter. +

Reviewer: Not sure if it's necessary to move + to a new line.

> Once you do this, the pages will automatically surface on the User Handbook home page (provided by link:content/doc/book/index.html.haml[`content/doc/book/index.html.haml`]), which will automatically appear https://jenkins.io/doc/book/[here] (and the TOC on the left of this page) when accepted. +

Reviewer: Move +?

> Once you do this, the pages will automatically surface on the User Handbook home page (provided by link:content/doc/book/index.html.haml[`content/doc/book/index.html.haml`]), which will automatically appear https://jenkins.io/doc/book/[here] (and the TOC on the left of this page) when accepted. +

Reviewer: ```suggestion
Once you do this, the pages automatically surface on the User Handbook home page (provided by link:content/doc/book/index.html.haml[`content/doc/book/index.html.haml`]), which will automatically appear https://jenkins.io/doc/book/[here] (and the TOC on the left of this page) when accepted. +
```

The content on this page should be predominantly "reference" material about the subject of page (e.g. more detailed information about a specific aspect of the big new feature).

> These pages may contain appropriate "discussion"- and "how-to guide"-like material (i.e. overviews and procedures) relevant to the subject to make the content more useful.

Reviewer: ```suggestion
These pages may contain appropriate "discussion" - and "how-to guide"- like material (i.e. overviews and procedures) relevant to the subject to make the content more useful.
```

. *Extend Jenkins Documentation:* This documentation is for people who want to _extend_ the functionality of Jenkins by developing their own Jenkins plugins.
Like the Jenkins User Documentation (above), the content is based on the same link:https://www.divio.com/blog/beginners_guide_to_documentation/[documentation model].
The content for this set of documentation is written up as a combination of `.haml` and `.adoc` files located in the link:content/doc/developer[`content/doc/developer/`] directory.
Read more about adding pages to this documentation in <<adding-a-stand-alone-page,Adding a stand-alone-page>>.
. *Solution pages:* Topic-specific destination pages providing a high-level overview of a topic with links into getting started guides, handbook chapters, relevant plugins, and multimedia related to the topic.
Be aware that some of this content might already be present in the Jenkins User / Extend Jenkins Documentation.

The documentation pages can use the same metadata (`title`, `description`, `opengraph:image`) as blog posts.

==== Adding images

> When you add screenshots or images to documentation, there are methods to ensure that the images are focused, clear, and useful to the reader:

Reviewer 2: Docs Office Hours (Asia) noted that it would be good to insert an example image markup into this text.  Alternately, link to the asciidoc documentation so that the writer can see multiple examples of image placement on the page, text wrapping, alt text, and how to implement the practices described in this guide.

Presenter: Kohsuke Kawaguchi
----

If using the GitHub UI to create this file, commit the file using the "Create a new branch for this commit and start a pull request" option.
If working via a local clone, commit the change, push to a branch, and start a PR as usual.

=== Adding a stand-alone page

Encouraged formats:

* link:https://asciidoctor.org[Asciidoctor] (basic content creation) (link:https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/[AsciiDoc syntax quick reference])
* link:http://haml.info[Haml] (more advanced/custom page) (link:http://haml.info/docs/yardoc/file.REFERENCE.html[Haml syntax reference])

Adding a new page is as easy as adding a new file to the link:content/[`content/`] directory.
It is important to keep in mind that the filename you choose *will be the URL of your page*, so ensure you have a *lowercase* and useful filename.

The link:content/index.html.haml[`content/index.html.haml`] page is one such example of a special-case, standalone page.

==== Clean URLs

> In order to have a clean URL, e.g. "https://jenkins.io/my-clean-url", you would need to create a directory with your content in it.

Reviewer: ```suggestion
To have a clean URL, e.g. "https://jenkins.io/my-clean-url", you would need to create a directory with your content in it.
```

Using the above example, I would create the directory `content/my-clean-url` and if I were creating an Asciidoc file, I would then create the file `content/my-clean-url/index.adoc`.
(Advanced Haml users would create `content/my-clean-url/index.html.haml`).

=== Adding a Logo

> In order to add a new logo, please submit a pull request, adding a new metadata `.yml` file in `content/_data/logo` and a new directory containing the logo assets into `content/images/logos/`.

Reviewer: ```suggestion
To add a new logo, submit a PR, adding a new metadata `.yml` file in `content/_data/logo` and a new directory containing the logo assets into `content/images/logos/`.
```

> Requirements for logos:

Reviewer: ```suggestion
Logo requirements:
```

* All submitted images are licensed under the link:https://creativecommons.org/licenses/by-sa/3.0/[Creative Commons Attribution-ShareAlike 3.0 Unported License].

> * At least 2 images are needed: full-size PNG and another PNG which has a 256px height.

Reviewer: ```suggestion
* At least two images are needed: full-size PNG and another PNG that has a 256px height.
```

=== Joining the teams

> If you are interested to join the Triage or Copy Editors team, you can request membership in the link:https://groups.google.com/d/forum/jenkinsci-dev[Jenkins Developer mailing list] or in the link:https://www.jenkins.io/sigs/docs/[Documentation SIG channels].

Reviewer: ```suggestion
If you are interested in joining the Triage or Copy Editors team, you can request membership in the link:https://groups.google.com/d/forum/jenkinsci-dev[Jenkins Developer mailing list] or in the link:https://www.jenkins.io/sigs/docs/[Documentation SIG channels].
```
